src/shared/udev-util.c

```diff
@@ -319,3 +322,50 @@ bool device_for_action(sd_device *dev, DeviceAction action) {
 
         return a == action;
 }
+
+int udev_rule_parse_value(char *str, char **ret_value, char **ret_endpos) {
+        char *value = str;
+        char *i, *j;
+        int r;
+        bool is_escaped;
+
+        /* value must be double quotated */
+        is_escaped = value[0] == 'e';
+        value += is_escaped;
+        if (value[0] != '"')
+                return -EINVAL;
+        value++;
+
+        if (!is_escaped) {
+                /* unescape double quotation '\"'->'"' */
+                for (i = j = value; *i != '"'; i++, j++) {
+                        if (*i == '\0')
+                                return -EINVAL;
+                        if (i[0] == '\\' && i[1] == '"')
+                                i++;
+                        *j = *i;
+                }
+                j[0] = '\0';
+        } else {
+                _cleanup_free_ char *unescaped = NULL;
+
+                /* find the end position of value */
+                for (i = value; *i != '"'; i++) {
+                        if (*i == '\0')
+                                return -EINVAL;
+                        if (is_escaped && i[0] == '\\' && i[1] != '\0')
```

Comment:
Hmm, first of all, `is_escaped` is always true here.

This loop must be done more carefully; the input may be `e"\\\\"`... Please try to add such a case in the test.